### PR TITLE
Improve resizing log message

### PR DESCRIPTION
**Pull request by georgwiese**
With this change, we check the fraction of used rows in each machine. If the fraction is above 50%, we don't log anything in INFO level, otherwise, we suggest that the machines could be configured with a smaller min_degree.

This is not a warning, because on some backends, it might not be possible to use VADCOP.

Example:
```
$ cargo run pil test_data/std/poseidon_gl_test.asm -o output -f
...
Only 101 of 256 rows (39.45%) are used in machine 'Main Machine', which is configured to be of static size 256. If the min_degree of this machine was lower, we could size it down such that the fraction of used rows is at least 50%. If the backend supports it, consider lowering the min_degree.
```

### CODE REVIEW

File: executor/src/witgen/machines/mod.rs [hunk 1]
```diff
@@ -214,3 +215,34 @@ impl<'a, T: FieldElement> MachineParts<'a, T> {
         self.fixed_data.column_name(poly_id)
     }
 }
+
+/// The minimum size for which a warning is logged if the used rows are less than half of the size.
+/// This number coincides with 2**powdr_linker::MIN_DEGREE_LOG.
```
Comment (chriseth) [inline executor/src/witgen/machines/mod.rs:220]: It doesn't because there is no static_assert.

Comment (chriseth) [reply executor/src/witgen/machines/mod.rs:220]: Is it important that it does?

File: executor/src/witgen/machines/mod.rs [hunk 2]
```diff
@@ -214,3 +215,34 @@ impl<'a, T: FieldElement> MachineParts<'a, T> {
...
+
+    if size > MIN_RERPORTING_SIZE && fraction_used < 0.5 {
+        // In a machine configured to use VADCOP, we would expect the next power of two to be used.
+        log::info!(
+            "Only {} / {} rows are used in machine {}, which is configured to support sizes in the range {}..{}. If the backend supports it, consider lowering the min_degree.",
+            used_rows,
+            size,
+            name,
+            degree_range.min,
+            degree_range.max,
```
[the remark this answers is not in the document]
Comment (chriseth) [reply executor/src/witgen/machines/mod.rs:237]: I think we should also say what the actual probleme here is. I'm not sure, but I guess that the machine could be sized down, but `min_degree` prevents the size being reduced?

File: executor/src/witgen/machines/mod.rs [hunk 3]
```diff
@@ -214,3 +215,34 @@ impl<'a, T: FieldElement> MachineParts<'a, T> {
...
+        );
+    } else {
+        log::debug!(
+            "{} / {} rows are used in machine {}.",
+            used_rows,
+            size,
+            name,
+        );
```
Comment (chriseth) [inline executor/src/witgen/machines/mod.rs:245]:
```suggestion
        log::debug!("{used_rows} of {size} rows are used in machine '{name}'.");
```

File: executor/src/witgen/machines/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (georgwiese) [reply executor/src/witgen/machines/mod.rs]: OK, added that! Also logging the percentage now and mention that the goal should be to be at least at 50%.

File: executor/src/witgen/machines/mod.rs (hunk 1 above)
Comment (georgwiese) [reply executor/src/witgen/machines/mod.rs:220]: It is not *really* important. If the total number of rows, the overhead of proving them becomes negligible and the log would just be noise. Arguably 32 is a already very small.
If this number was below `2**powdr_linker::MIN_DEGREE_LOG`, we'd get the log message often, that's why I chose it to be equal.

Yesterday, we also discussed with Thibaut and Leo that we want to remove `MIN_DEGREE_LOG` and let the user always be explicit about the min degree. Then, this comment goes away.

**Comment by leonardoalt** [comment]
can we merge this?

**Review by chriseth — APPROVED** [review]

**Pull request merged**